storage/innobase/btr/btr0sea.cc

```diff
@@ -643,7 +643,7 @@ btr_search_update_hash_ref(
 				rec_get_offsets(rec, index, offsets_,
 						ULINT_UNDEFINED, &heap),
 				block->curr_n_fields,
-				block->curr_n_bytes, index->id);
+				block->curr_n_bytes, index);
```

Comment:
Yes, it make sense.
But, I think it is not necessary to always carefully distinguish between clustered and secondary indexes for rec_*(). 
Although rec_fold () is only used for the adaptive hash index now, for secondary index only. 
Is it possible to change it?
Otherwise, we should add a assert(!dict_index_is_cluster(dict_load_index_by_id(index_id)); in rec_fold().